### ISSUE TITLE
Test::Version escapes special characters in file names incorrectly

### ISSUE THREAD

**Issue by van-de-bugger** [plicease/Dist-Zilla-Plugin-Test-Version#8]
Consider example:

```
name     = Assa
version  = 0.001
abstract = Abstract
[GenerateFile]
    filename = lib/As\'sa.pm
    content  = package Assa;
    content  = our $VERSION = '0.001';
    content  = 1;
[Test::Version]
    finder = :InstallModules
```

`dzil build` completes successfully. `GenerateFile` generates file `lib/As\'sa.pm`, as specified. However, `Test::Version` generates invalid Perl code in `xt/author/test-version.t`:

```
...
version_ok('lib/As\\'sa.pm');
...
```

Note: `'lib/As\\'sa.pm'` is not valid Perl single-quoted string.

The problem is in this code:

```
map { s/'/\\'/g; "version_ok('$_');" } @filenames
```

Escaping apostrophes with backslashes is not enough. All the existing backslashes should be escaped (i. e. doubled) too.

**Comment by simbabque** [comment]
I'm looking at this as part of the Pull Request Challenge 2017. It's pretty trivial to add the code that does this, but I don't think it will make sense. 

I've created a test, but  Module::Metadata (which is used by Test::Version) is not able to extract the package name `Assa` from a file `As\'sa.pm`. Maybe this still worked back in 2015, but the most current version cannot do that.

So even if we generate a syntactically correct .t file, the Test::Version test inside it will fail because there is no version if the package name does not match the .pm file name.

Did you actually encounter this problem in the wild, or is it constructed?

**Comment by plicease** [comment]
I don't believe this this problem exists in the wild.  I've left this issue open to signal a willingness to accept a patch, but it isn't a priority for me personally.
